File: code/java/checkPossibility.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        if(nums.length <= 2)
            return true;
        for(int i=0;i<nums.length-1;i++){
            if(nums[i] > nums[i+1])
                return helper(nums,i,i+1) || helper(nums,i+1,i);
        }
        return true;
    }
    
    public boolean helper(int[] nums,int k,int j){
        int tmp = nums[k];
        nums[k] = nums[j];
        for(int i=0;i<nums.length-1;i++){
            if(nums[i]>nums[i+1]){
                nums[k] = tmp;
                return false;
            }
        }
        return true;
    }
}

File: code/python/checkPossibility.py
class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        if len(nums) <= 2:
            return True
        for i in range(len(nums)-1):
            if nums[i] > nums[i+1]:
                return self.helper(nums,i,i+1) or self.helper(nums,i+1,i)
        return True
    
    def helper(self,nums,k,j)->bool:
        temp = nums[j]
        nums[j] = nums[k]
        for i in range(len(nums)-1):
            if nums[i] > nums[i+1]:
                nums[j] = temp
                return False
        return True
                
            
